src/data-loader.ts

```
@@ -687,24 +735,38 @@ export async function loadDailyUsageData(
 	}
 
-	// Group by date using Object.groupBy
-	const groupedByDate = groupBy(allEntries, entry => entry.date);
+	// Group by date, optionally including project
+	// Automatically enable project grouping when project filter is specified
+	const needsProjectGrouping = options?.groupByProject === true || options?.project != null;
+	const groupingKey = needsProjectGrouping
+		? (entry: typeof allEntries[0]) => `${entry.date}|${entry.project}`
+		: (entry: typeof allEntries[0]) => entry.date;
+
+	const groupedData = groupBy(allEntries, groupingKey);
```

ESLint

[error] Unsafe assignment of an error typed value.

(ts/no-unsafe-assignment)